ThunderCats/UIColor+TCAdditions.m

```diff
@@ -28,12 +28,19 @@
 
 
 #import "UIColor+TCAdditions.h"
+#import "TCInvalidArgument.h"
 
 @implementation UIColor (TCAdditions)
 
 + (UIColor *)tc_colorWithHexString:(NSString *)hexString
 {
     NSString *cleanString = [hexString stringByReplacingOccurrencesOfString:@"#" withString:@""];
+    
+    if ([cleanString length] != 3 && [cleanString length] != 6)
+    {
+        [TCInvalidArgument raiseWithReason:@"invalid hex string"];
+    }
+    
```

Comment:
So, are you saying that we should raise an exception and not let '##0#123#4#5##' through?